src/components/dialogs/generator-modification-dialog.js

```diff
@@ -294,6 +301,17 @@ const GeneratorModificationDialog = ({
             generatorInfos?.minMaxReactiveLimits?.minimumReactivePower,
     });
 
+    useEffect(() => {
+        setReactivePowerRequired(
+            (minimumReactivePower !== '' &&
+                !generatorInfos?.minMaxReactiveLimits?.minimumReactivePower) ||
+                (maximumReactivePower !== '' &&
+                    !generatorInfos?.minMaxReactiveLimits?.maximumReactivePower)
+                ? true
+                : undefined // if the field is not required then we set "reactivePowerRequired" to undefined so that the optional label is not displayed
```

Comment:
That's tricky. We used this in other places ?
But maybe it's not a big deal as we'll refacto forms.